independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/DirectoryPathTree.java

```diff
@@ -10,11 +11,27 @@
 import java.util.Objects;
 import java.util.function.Consumer;
 import java.util.function.Function;
+import java.util.regex.Pattern;
 
 public class DirectoryPathTree extends PathTreeWithManifest implements OpenPathTree, Serializable {
 
     private static final long serialVersionUID = 2255956884896445059L;
 
+    private static final boolean USE_WINDOWS_ABSOLUTE_PATH_PATTERN = !FileSystems.getDefault().getSeparator().equals("/");
+
+    private static volatile Pattern windowsAbsolutePathPattern;
+
+    private static Pattern windowsAbsolutePathPattern() {
+        return windowsAbsolutePathPattern == null ? windowsAbsolutePathPattern = Pattern.compile("[a-zA-Z]:\\\\.*")
```

Comment:
Are we sure this is comprehensive ? UNC formats are also considered valid "absolute paths" on Windows; some people might even argue that URL is the golden standard.

Reply:
(I'm fine either way - just wondering if this was your intent. Perhaps a doc clarification of what's intended is in order)

Reply:
That would be nice if we wanted to support `PathTree` as a general purpose API, which isn't the case currently. For now it's limited to loading resources on the classpath following the Java classpath resource name format. I'd say the main goal currently is to prevent reading outside the directory tree and this seems to be enough. Actually, this check for absolute path is already beyond that and could probably be removed.